lib/rubocop/cop/lint/unneeded_require_statement.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Checks for unnecessary require statement.
+      #
+      # The following features are unnecessary require statement because
+      # they are already loaded.
+      #
+      # ruby -ve 'p $LOADED_FEATURES.reject { |feature| %r|/| =~ feature }'
+      # ruby 2.2.8p477 (2017-09-14 revision 59906) [x86_64-darwin13]
+      # ["enumerator.so", "rational.so", "complex.so", "thread.rb"]
+      #
+      # This cop targets Ruby 2.2 or higher containing these 4 features.
+      #
+      # @example
+      #   # bad
+      #   require 'unloaded_feature'
+      #   require 'thread'
+      #
+      #   # good
+      #   require 'unloaded_feature'
+      class UnneededRequireStatement < Cop
+        MSG = 'Remove unnecessary require statement.'.freeze
+
+        def_node_matcher :unnecessary_require_statement?, <<-PATTERN
+          (send nil :require
+            (str {"enumerator" "rational" "complex" "thread"}))
+        PATTERN
+
+        def on_send(node)
+          return if RUBY_VERSION < '2.2'
```

Comment:
Do not use `RUBY_VERSION` to check ruby version for "parsing". `RUBY_VERSION` is a runtime ruby version.
Please use `minimum_target_ruby_version 2.2` instead of.

Then test will be broken. The test will be fixed by this patch.

```diff
diff --git a/lib/rubocop/cop/lint/unneeded_require_statement.rb b/lib/rubocop/cop/lint/unneeded_require_statement.rb
index 00b1944d..cfcd8417 100644
--- a/lib/rubocop/cop/lint/unneeded_require_statement.rb
+++ b/lib/rubocop/cop/lint/unneeded_require_statement.rb
@@ -22,6 +22,10 @@ module RuboCop
       #   # good
       #   require 'unloaded_feature'
       class UnneededRequireStatement < Cop
+        extend TargetRubyVersion
+
+        minimum_target_ruby_version 2.2
+
         MSG = 'Remove unnecessary require statement.'.freeze
 
         def_node_matcher :unnecessary_require_statement?, <<-PATTERN
@@ -30,7 +34,6 @@ module RuboCop
         PATTERN
 
         def on_send(node)
-          return if RUBY_VERSION < '2.2'
           return unless unnecessary_require_statement?(node)
           add_offense(node)
         end
diff --git a/spec/rubocop/cop/lint/unneeded_require_statement_spec.rb b/spec/rubocop/cop/lint/unneeded_require_statement_spec.rb
index bafbfa09..00b2ea86 100644
--- a/spec/rubocop/cop/lint/unneeded_require_statement_spec.rb
+++ b/spec/rubocop/cop/lint/unneeded_require_statement_spec.rb
@@ -1,10 +1,8 @@
 # frozen_string_literal: true
 
-describe RuboCop::Cop::Lint::UnneededRequireStatement do
-  let(:config) { RuboCop::Config.new }
+describe RuboCop::Cop::Lint::UnneededRequireStatement, :config do
   subject(:cop) { described_class.new(config) }
 
-  if RUBY_VERSION < '2.2'
     context 'target ruby version < 2.2', :ruby21 do
       it "does not registers an offense when using `require 'enumerator'`" do
         expect_no_offenses(<<-RUBY.strip_indent)
@@ -24,7 +22,6 @@ describe RuboCop::Cop::Lint::UnneededRequireStatement do
         RUBY
       end
     end
-  else
     context 'target ruby version >= 2.2', :ruby22 do
       it "registers an offense when using `require 'enumerator'`" do
         expect_offense(<<-RUBY.strip_indent)
@@ -44,5 +41,4 @@ describe RuboCop::Cop::Lint::UnneededRequireStatement do
         RUBY
       end
     end
-  end
 end
```

Reply:
Thanks for the review. You're right, this is my mistake. I will fix it along with other review comments pointed out.

Reply:
I fixed it. And in addition, I fixed an error of spec code in this PR. 73b01db